tests/contest/runtimetest/src/utils.rs

```diff
@@ -471,3 +472,37 @@ pub fn test_mount_rsuid_option(path: &str) -> Result<(), std::io::Error> {
         format!("rsuid error {path:?}"),
     ))
 }
+
+pub fn test_additional_gids(gids: &Vec<u32>) -> Result<(), std::io::Error> {
```

Comment:
I think we can move this to the test file itself, I don't think we will be using this anywhere else.
Also please use safe alternatives from nix instead of unsafe from libc.
Can you add comments explaining what this is trying to do? The original go does not have comments, so it would be good to have explanation here.